### PR TITLE
Run merge operation on handler in Search threadpool

### PR DESCRIPTION
**Pull request by mfussenegger**
Some sys.snapshots queries failed occasionally because there is a new
assertion that verifies that operations on the `BlobStoreRepository`
happen in either Generic, Snapshot or Search.

 - [x] User relevant changes are recorded in ``CHANGES.txt``
 - [x] Touched code is covered by tests
 - [x] Documentation has been updated if necessary
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

File: sql/src/main/java/io/crate/execution/engine/InterceptingRowConsumer.java [hunk 1]
```diff
@@ -78,7 +82,7 @@ private void tryForwardResult(Throwable throwable) {
         }
         if (failure == null) {
             assert iterator != null : "iterator must be present";
-            consumer.accept(iterator, null);
+            executor.execute(() -> consumer.accept(iterator, null));
```
Comment (andreidan) [inline sql/src/main/java/io/crate/execution/engine/InterceptingRowConsumer.java]: should we handle the rejectedexecutionexception exceptions here? (maybe even use our `CompletableFutures.supplyAsync`)

File: sql/src/main/java/io/crate/expression/reference/sys/snapshot/SysSnapshots.java [hunk 2]
```diff
@@ -21,66 +21,39 @@
...
+                        snapshotInfo.version().toString(),
+                        snapshotInfo.state().name()
                     );
-                }
-            }));
-        }
-        return sysSnapshots;
+                })
+            );
+        return stream::iterator;
```
Comment (andreidan) [inline sql/src/main/java/io/crate/expression/reference/sys/snapshot/SysSnapshots.java]: 👍

File: sql/src/main/java/io/crate/execution/engine/InterceptingRowConsumer.java (hunk 1 above)
Comment (mfussenegger) [reply sql/src/main/java/io/crate/execution/engine/InterceptingRowConsumer.java]: See https://github.com/crate/crate/pull/7853/files#diff-dd02b03e4e7ce6a1e8bb0edec0bde6ceR66 - the Executor has a fallback. 
Maybe that's a bit too subtle. I can also change it to have explicit handling here?

Comment (mfussenegger) [reply sql/src/main/java/io/crate/execution/engine/InterceptingRowConsumer.java]: Merged it for now, I can follow up with another PR to make it explicit if you want, but for now I'd like to get the flakyness away to unblock the other PRs

Comment (andreidan) [reply sql/src/main/java/io/crate/execution/engine/InterceptingRowConsumer.java]: Ah right, that's fine

**Pull request merged**